compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -69,6 +69,11 @@ pub(crate) fn eval_nullary_intrinsic<'tcx>(
             ty::Alias(..) | ty::Param(_) | ty::Placeholder(_) | ty::Infer(_) => {
                 throw_inval!(TooGeneric)
             }
+            ty::Pat(..) => {
+                unimplemented!(
```

Comment:
If the only thing that is allowed is range patterns, then the scrutinee is never something with variants? Unless you mean an enum, in which case does something like:

```
#[repr(u8)] enum Two { A, B }
type MoreThanTwo = Two is 0..10u8;
```

Not cause problems with `variant_count::<MoreThanTwo>()`? What does it evaluate to?

Reply:
yea, I am only worried about other pattern types. I'll exhaustively match and comment as much